demo-app/src/app/pagination/pagination.component.html

```diff
@@ -0,0 +1,195 @@
+    </p>
+</div>
+
+<div class="section">
+  <h5 class="light">Active page</h5>
+
+  <div class="row m-valign-wrapper">
+    <mz-pagination class="col s12 m6"
+      [currentPage]="5"
+      [itemsPerPage]="10"
+      [totalItems]="100"
+    >
+    </mz-pagination>
+
+    <p class="col s12 m6">
+      Use <code class="language-markup">currentPage</code> input property to select a specific page when rendering.
+    </p>
+</div>
+
+<div class="section">
+  <h5 class="light">Displayed pages</h5>
+
+  <div class="row m-valign-wrapper">
+    <mz-pagination class="col s12 m6"
+      [itemsPerPage]="10"
+      [maxPageButtons]="10"
+      [totalItems]="100"
+    >
+    </mz-pagination>
+
+    <p class="col s12 m6">
+      Optionally, you can change the number of displayed pages by using <code class="language-markup">maxPageButton</code> proprety. When no value is provided the number of displayed pages is defaulted to 5.
+    </p>
+</div>
+
+<div class="section">
+  <h5 class="light">Basic pagination</h5>
+
+  <div class="row m-valign-wrapper">
+    <mz-pagination class="col s12 m6"
+      [itemsPerPage]="10"
+      [totalItems]="100"
+    >
+    </mz-pagination>
+
+    <p class="col s12 m6">
+      By setting <code class="language-markup">itemsPerPage</code> and <code class="language-markup">totalItems</code> input property on <code class="language-markup">mz-pagination</code> component, the number of available pages will automatically be calculated.
+    </p>
+</div>
+
+<div class="section">
+  <h5 class="light">First and last page button</h5>
+
+  <div class="row m-valign-wrapper">
+    <mz-pagination class="col s12 m6"
+      [enableFirstAndLastPageButtons]="true"
+      [itemsPerPage]="10"
+      [totalItems]="100"
+    >
+    </mz-pagination>
+
+    <p class="col s12 m6">
+      You can show first page and last page buttons by setting <code class="language-markup">enableFirstAndLastButtons</code> input property to <code class="language-markup">true</code>.
+    </p>
+</div>
+
+<div class="section">
+  <h5 class="light">Listen to page change</h5>
+
+  <div class="row m-valign-wrapper">
+    <mz-pagination class="col s12 m6"
+      (pageChanged)="pageChange($event)"
+      [itemsPerPage]="10"
+      [totalItems]="100"
+    >
+    </mz-pagination>
+
+    <p class="col s12 m6">
+      You can use the ouput property <code class="language-markup">changePage</code> to call a function when page changed.
+    </p>
+</div>
+
+<div class="section">
+  <h5 class="light">Playground</h5>
+
+  <p>
+    The input properties of the <code class="language-markup">mz-pagination</code> directive makes the form-control completely dynamic. Change the configuration below and see how the checkbox reacts.
+  </p>
+
+  <div class="row">
+
+    <div class="col s12 m6">
+      <mz-pagination
+        [currentPage]="paginationCurrentPage"
```

Comment:
I would expect the `currentPage` to change its value when I am changing page...

![pagination-playground](https://user-images.githubusercontent.com/6987084/36644453-b6893536-1a28-11e8-8d12-fd9cd522a16d.gif)